meta-ros1-melodic/recipes-bbappends/mavlink/mavlink_2019.11.11-1.bbappend

```diff
@@ -0,0 +1,16 @@
+ROS_BUILD_DEPENDS += " \
+    ${PYTHON_PN}-future-native \
+"
+
+# Doesn't inherit ros_catkin, only ros_cmake so we need to inherit right python*native here
```

Comment:
Should ROS_BUILD_TYPE be changed?

Reply:
It builds fine with cmake, it's just the inherit in ros_catkin.bbclass:
inherit ${@'distutils3-base' if d.getVar('ROS_PYTHON_VERSION', True) == '3' else 'distutils-base'}

which inherits the right pythonnative for other recipes.

Reply:
Well with catkin it will respect the opt-prefix, with cmake it installs in regular usr:

Only in tmp-glibc/work/aarch64-oe-linux/mavlink/2020.2.2-1-r0-catkin/image: opt
Only in tmp-glibc/work/aarch64-oe-linux/mavlink/2020.2.2-1-r0-cmake/image: usr

```
$ diff -rq 2020.2.2-1-r0-cmake/image/usr/include/ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/include/
Files 2020.2.2-1-r0-cmake/image/usr/include/mavlink/v1.0/common/common.h and 2020.2.2-1-r0-catkin/image/opt/ros/melodic/include/mavlink/v1.0/common/common.h differ
$ diff -rq 2020.2.2-1-r0-cmake/image/usr/lib/ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/lib/
Files 2020.2.2-1-r0-cmake/image/usr/lib/pkgconfig/mavlink.pc and 2020.2.2-1-r0-catkin/image/opt/ros/melodic/lib/pkgconfig/mavlink.pc differ
$ diff -rq 2020.2.2-1-r0-cmake/image/usr/share/ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/share/
Files 2020.2.2-1-r0-cmake/image/usr/share/mavlink/cmake/mavlink-config.cmake and 2020.2.2-1-r0-catkin/image/opt/ros/melodic/share/mavlink/cmake/mavlink-config.cmake differ
```

Looks like using catkin would be better, @yossizap what's your opinion?

Reply:
The lib, share diffs are just the path change
```
$ diff -uNr 2020.2.2-1-r0-cmake/image/usr/lib/ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/lib/
diff -uNr 2020.2.2-1-r0-cmake/image/usr/lib/pkgconfig/mavlink.pc 2020.2.2-1-r0-catkin/image/opt/ros/melodic/lib/pkgconfig/mavlink.pc
--- 2020.2.2-1-r0-cmake/image/usr/lib/pkgconfig/mavlink.pc      2020-02-12 09:03:31.000000000 -0800
+++ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/lib/pkgconfig/mavlink.pc 2020-02-12 08:59:09.000000000 -0800
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=/usr
+prefix=/opt/ros/melodic
+exec_prefix=/opt/ros/melodic
 includedir=${prefix}/include

 Name: mavlink
```
```
$ diff -uNr 2020.2.2-1-r0-cmake/image/usr/share/ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/share/
diff -uNr 2020.2.2-1-r0-cmake/image/usr/share/mavlink/cmake/mavlink-config.cmake 2020.2.2-1-r0-catkin/image/opt/ros/melodic/share/mavlink/cmake/mavlink-config.cmake
--- 2020.2.2-1-r0-cmake/image/usr/share/mavlink/cmake/mavlink-config.cmake      2020-02-12 09:03:31.000000000 -0800
+++ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/share/mavlink/cmake/mavlink-config.cmake 2020-02-12 08:59:09.000000000 -0800
@@ -21,7 +21,7 @@
 foreach(lib )
   set(onelib "${lib}-NOTFOUND")
   find_library(onelib ${lib}
-      PATHS "/usr/lib"
+      PATHS "/opt/ros/melodic/lib"
     NO_DEFAULT_PATH
     )
   if(NOT onelib)
```

But the include/mavlink/v1.0/common/common.h diff is a bit surprising:
```
$ diff -uNr 2020.2.2-1-r0-cmake/image/usr/include/ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/include/
diff -uNr 2020.2.2-1-r0-cmake/image/usr/include/mavlink/v1.0/common/common.h 2020.2.2-1-r0-catkin/image/opt/ros/melodic/include/mavlink/v1.0/common/common.h
--- 2020.2.2-1-r0-cmake/image/usr/include/mavlink/v1.0/common/common.h  2020-02-12 09:03:32.000000000 -0800
+++ 2020.2.2-1-r0-catkin/image/opt/ros/melodic/include/mavlink/v1.0/common/common.h     2020-02-12 08:59:10.000000000 -0800
@@ -555,165 +555,6 @@
 } ORBIT_YAW_BEHAVIOUR;
 #endif

-/** @brief Commands to be executed by the MAV. They can be executed on user request, or as part of a mission script. If the action is used in a mission, the parameter mapping to the waypoint/mission message is as follows: Param 1, Param 2, Param 3, Param 4, X: Param 5, Y:Param 6, Z:Param 7. This command list is similar what ARINC 424 is for commercial aircraft: A data format how to interpret waypoint/mission data. See https://mavlink.io/en/guide/xml_schema.html#MAV_CMD for information about the structure of the MAV_CMD entries */
-#ifndef HAVE_ENUM_MAV_CMD
-#define HAVE_ENUM_MAV_CMD
-typedef enum MAV_CMD
-{
-   MAV_CMD_NAV_WAYPOINT=16, /* Navigate to waypoint. |Hold time. (ignored by fixed wing, time to stay at waypoint for rotary wing)| Acceptance radius (if the sphere with this radius is hit, the waypoint counts as reached)| 0 to pass through the WP, if > 0 radius to pass by WP. Positive value for clockwise orbit, negative value for counter-clockwise orbit. Allows trajectory control.| Desired yaw angle at waypoint (rotary wing). NaN to use the current system yaw heading mode (e.g. yaw towards next waypoint, yaw to home, etc.).| Latitude| Longitude| Altitude|  */
-   MAV_CMD_NAV_LOITER_UNLIM=17, /* Loiter around this waypoint an unlimited amount of time |Empty| Empty| Radius around waypoint. If positive loiter clockwise, else counter-clockwise| Desired yaw angle. NaN to use the current system yaw heading mode (e.g. yaw towards next waypoint, yaw to home, etc.).| Latitude| Longitude| Altitude|  */
-   MAV_CMD_NAV_LOITER_TURNS=18, /* Loiter around this waypoint for X turns |Number of turns.| Empty| Radius around waypoint. If positive loiter clockwise, else counter-clockwise| Forward moving aircraft this sets exit xtrack location: 0 for center of loiter wp, 1 for exit location. Else, this is desired yaw angle. NaN to use the current system yaw heading mode (e.g. yaw towards next waypoint, yaw to home, etc.).| Latitude| Longitude| Altitude|  */
...
-   MAV_CMD_USER_3=31012, /* User defined command. Ground Station will not show the Vehicle as flying through this item. Example: MAV_CMD_DO_SET_PARAMETER item. |User defined| User defined| User defined| User defined| User defined| User defined| User defined|  */
-   MAV_CMD_USER_4=31013, /* User defined command. Ground Station will not show the Vehicle as flying through this item. Example: MAV_CMD_DO_SET_PARAMETER item. |User defined| User defined| User defined| User defined| User defined| User defined| User defined|  */
-   MAV_CMD_USER_5=31014, /* User defined command. Ground Station will not show the Vehicle as flying through this item. Example: MAV_CMD_DO_SET_PARAMETER item. |User defined| User defined| User defined| User defined| User defined| User defined| User defined|  */
-   MAV_CMD_ENUM_END=31015, /*  | */
-} MAV_CMD;
-#endif
-
 /** @brief A data stream is not a fixed set of messages, but rather a
      recommendation to the autopilot software. Individual autopilots may or may not obey
      the recommended messages. */
```

Reply:
In that case, I prefer catlink to have it in the appropriate /opt/ros/ folder.

Reply:
Will update it to use catkin ROS_BUILD_TYPE but maybe we will need to eventually change how  ros_opt_prefix is applied for it to be respected by ros_cmake.bbclass as well, because there are 56 recipes just in melodic which use just plain cmake without catkin as ROS_BUILD_TYPE (only a few compared with 1635 using catkin), but still there might be some valid reason why
https://github.com/mavlink/mavlink-gbp-release/blob/release/melodic/mavlink/2019.11.11-1/package.xml#L29
says cmake and not catkin